src/log_batch.rs

```diff
@@ -410,6 +409,38 @@ impl LogItemBatch {
         }
     }
 
+    /// Prepare the `write` by signing a checksum, so-called `signature`,
+    /// into the `LogBatch`.
+    ///
+    /// The `signature` is both generated by the given `LogFileContext`.
+    /// That is, the final checksum of each `LogBatch` consists of this
+    /// `signature` and the original `checksum` of the contents.
+    pub(crate) fn prepare_write(buf: &mut Vec<u8>, file_context: &LogFileContext) -> Result<()> {
+        if !buf.is_empty() {
+            if let Some(signature) = file_context.get_signature() {
+                // Insert the signature into the encoded bytes. Rewrite checksum of
+                // `LogItemBatch` in `LogBatch`.
+                let footer_checksum_offset = buf.len() - LOG_BATCH_CHECKSUM_LEN;
+                match codec::decode_u32_le(&mut &buf[footer_checksum_offset..]) {
```

Comment:
This line probably won't fail. No need to encode extra context information.
```suggestion
                codec::decode_u32_le(&mut &buf[footer_checksum_offset..])?;
```